libyul/AST.h

```diff
@@ -35,56 +35,81 @@ namespace solidity::yul
 
 using Type = YulString;
 
-struct TypedName { langutil::SourceLocation location; YulString name; Type type; };
+struct DebugData
+{
+	explicit DebugData(langutil::SourceLocation _location): location(std::move(_location)) {}
+	langutil::SourceLocation location;
+};
```

Comment:
Do you really need a constructor here? Even if you don't provide one, you will still be able to do `DebugData{location}` to initialize the struct.

Reply:
`make_shared<DebugData const>(location)` doesn't work without constructor. I can use 
`make_shared<DebugData const>(DebugData{move(location)})`
I think it is better to have constructor, what do you think?

Reply:
Oh. In that case I'm fine with having a constructor. I only pointed it out because for structs it's generally not needed (unless you want to do something more than just fill in the fields) but looks like you have a genuine need for it here.